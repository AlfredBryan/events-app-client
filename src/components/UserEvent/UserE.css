.none-yet {
  background: #6ec7e3;
  width: 947px;
  height: 300px;
  margin-top: 10px;
}

.links a {
  text-align: center;
  text-decoration: none;
  color: red;
}

.none-yet a {
  text-decoration: none;
  color: red;
}

.main-sign-up {
  width: 947px;
  height: 877px;
  background: #6ec7e3;
  box-shadow: 0px 4px 0px rgba(0, 0, 0, 0.21);
  display: inline-block;
}

* {
  box-sizing: border-box;
}

.main-sign-up h3 {
  margin-top: 40px;
}

input[type="text"],
select,
textarea {
  width: 100%;
  padding: 12px;
  border: 1px solid #ccc;
  resize: vertical;
  border-radius: 10px;
}

label {
  padding: 12px 12px 12px 0;
  display: inline-block;
  font-family: Montserrat;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  font-size: 18px;
}

input[type="submit"] {
  background: #90b2bd;
  border: none;
  border-radius: 27px;
  cursor: pointer;
  margin-top: 100px;
  width: 278px;
  height: 76px;
  font-family: Montserrat;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  font-size: 24px;
  margin-left: 340px;
}

input[type="submit"]:hover {
  background-color: #757f83;
}

.form-container {
  border-radius: 5px;
  padding: 20px;
}

.col-25 {
  float: left;
  width: 150px;
  margin-top: 15px;
}

.col-75 {
  float: left;
  width: 641px;
  margin-top: 15px;
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

/* Responsive layout - when the screen is less than 600px wide, make the two columns stack on top of each other instead of next to each other */
@media screen and (max-width: 500px) {
  .col-25,
  .col-75,
  input[type="submit"] {
    width: 100%;
    margin-top: 0;
  }
  input[type="submit"] {
    margin-top: 30px;
    border-radius: 0%;
    margin-left: 0px;
  }
  .main-sign-up {
    width: 380px;
    margin-top: 15px;
  }
}

@media (max-width: 500px) {
  .none-yet {
    width: 380px;
    margin-top: 20px;
  }
}
