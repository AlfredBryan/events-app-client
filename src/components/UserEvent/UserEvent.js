import React, { Component } from "react";
import axios from "axios";

import NavBar from "../NavBar/NavBar";

const token = localStorage.getItem("token");
const apiURL = "https://events-apps.herokuapp.com/api/events";

class UserEvent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      eventId: this.props.match.params,
      name: "",
      address: "",
      phone: "",
      location: "",
      reason: "",
      errorMessage: ""
    };
  }

  parseJwt = token => {
    if (!token) {
      return;
    }
    const base64Url = token.split(".")[1];
    const base64 = base64Url.replace("-", "+").replace("_", "/");
    return JSON.parse(window.atob(base64));
  };

  handleSubmit = e => {
    e.preventDefault();
    let user = this.parseJwt(token);
    let userId = user.id;
    let id = this.state.eventId.id;
    let { name, address, phone, location, reason } = this.state;
    axios
      .post(`${apiURL}/${id}`, {
        userId,
        name,
        address,
        phone,
        location,
        reason
      })
      .then(res => {
        if (res.status === 201) {
          this.props.history.replace("/home");
        } else {
          this.setState({
            errorMessage: "User already signed up"
          });
        }
      });
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  render() {
    return (
      <React.Fragment>
        <NavBar />
        <div className="wrapper">
          <div className="main-sign-up">
            <h3>Event Sign Up</h3>
            <p style={{ textAlign: "center" }}>{this.state.errorMessage}</p>
            <div className="form-container">
              <form encType="multipart/form-data" onSubmit={this.handleSubmit}>
                <div className="row">
                  <div className="col-25">
                    <label htmlFor="name">Name of User</label>
                  </div>
                  <div className="col-75">
                    <input
                      type="text"
                      id="name"
                      name="name"
                      placeholder="Your name.."
                      value={this.state.name}
                      onChange={this.handleChange}
                    />
                  </div>
                </div>
                <div className="row">
                  <div className="col-25">
                    <label htmlFor="address">Address</label>
                  </div>
                  <div className="col-75">
                    <input
                      type="text"
                      id="address"
                      name="address"
                      placeholder="Your Address..."
                      value={this.state.address}
                      onChange={this.handleChange}
                    />
                  </div>
                </div>
                <div className="row">
                  <div className="col-25">
                    <label htmlFor="phone">Phone Number</label>
                  </div>
                  <div className="col-75">
                    <input
                      type="text"
                      id="phone"
                      name="phone"
                      placeholder="Enter Phone Number..."
                      value={this.state.phone}
                      onChange={this.handleChange}
                    />
                  </div>
                </div>
                <div className="row">
                  <div className="col-25">
                    <label htmlFor="location">Location</label>
                  </div>
                  <div className="col-75">
                    <input
                      type="text"
                      id="location"
                      name="location"
                      placeholder="Enter Location..."
                      value={this.state.location}
                      onChange={this.handleChange}
                    />
                  </div>
                </div>
                <div className="row">
                  <div className="col-25">
                    <label htmlFor="reason">Reason for signup</label>
                  </div>
                  <div className="col-75">
                    <textarea
                      id="reason"
                      name="reason"
                      placeholder="Write something.."
                      style={{ height: "152px" }}
                      value={this.state.reason}
                      onChange={this.handleChange}
                    />
                  </div>
                </div>
                <div className="row">
                  <input
                    onClick={this.handleSubmit}
                    type="submit"
                    value="Submit"
                  />
                </div>
              </form>
            </div>
          </div>
          <div className="footer" />
        </div>
      </React.Fragment>
    );
  }
}

export default UserEvent;
