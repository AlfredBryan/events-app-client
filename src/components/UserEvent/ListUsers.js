import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";

import "./UserE.css";
import NavBar from "../NavBar/NavBar";

const apiURL = "http://localhost:4000/api/events";

class ListUsers extends Component {
  constructor(props) {
    super(props);
    this.state = {
      eventId: this.props.match.params,
      users: []
    };
  }

  componentDidMount() {
    let eventId = this.state.eventId;
    let id = eventId.id;
    axios.get(`${apiURL}/${id}`).then(res => {
      this.setState({
        users: res.data.Users
      });
    });
  }

  render() {
    let { users } = this.state;
    let { eventId } = this.state;
    let id = eventId.id;
    if (users) {
      return (
        <React.Fragment>
          <NavBar />
          <div className="user-main links">
            <div className="container">
              <h5>Users Signed Up to Event</h5>
              <Link to={`/event/signup/${id}`}>
                <p>click to signup for event</p>
              </Link>
              {users.map(user => (
                <div key={user.id} className="d-flex wrap">
                  <p className="fname">{user.firstName}</p>
                  <p className="lname">{user.lastName}</p>
                </div>
              ))}
            </div>
          </div>
        </React.Fragment>
      );
    } else {
      return (
        <React.Fragment>
          <NavBar />
          <div className="container none-yet">
            <div className="row">
              <div className="col-10 mx-auto text-center text-title pt-5">
                <h2>No SignUp to Event Yet</h2>
                <Link to={`/event/signup/${id}`}>
                  <p>click to signup for event</p>
                </Link>
              </div>
            </div>
          </div>
        </React.Fragment>
      );
    }
  }
}

export default ListUsers;
