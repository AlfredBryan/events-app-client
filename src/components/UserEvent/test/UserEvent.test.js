import React from "react";
import { shallow } from "enzyme";
import UserEvent from "../UserEvent";

const match = {
  params: {
    eventId: 678256526 //any id you want to set
  }
};

describe("UserEvent Component", () => {
  it("should render without throwing an error", () => {
    expect(
      shallow(<UserEvent match={match} />)
        .find("form.userevent")
        .exists()
    ).toMatchSnapshot();
  });
  it("renders Name input", () => {
    expect(
      shallow(<UserEvent match={match} />).find("#name").length
    ).toMatchSnapshot();
  });
  it("renders Address input", () => {
    expect(
      shallow(<UserEvent match={match} />).find("#address").length
    ).toMatchSnapshot();
  });
  it("renders Phone input", () => {
    expect(
      shallow(<UserEvent match={match} />).find("#phone").length
    ).toMatchSnapshot();
  });
  it("renders Location input", () => {
    expect(
      shallow(<UserEvent match={match} />).find("#location").length
    ).toMatchSnapshot();
  });
  it("renders Reason input", () => {
    expect(
      shallow(<UserEvent match={match} />).find("#reason").length
    ).toMatchSnapshot();
  });
});

describe("Name input", () => {
  it("should respond to change event and change the state of the UserEvent Component", () => {
    const wrapper = shallow(<UserEvent match={match} />);
    wrapper.find("#name").simulate("change", {
      target: { name: "name", value: "test name" }
    });

    expect(wrapper.state("name")).toMatchSnapshot("test name");
  });
});

describe("Address input", () => {
  it("should respond to change event and change the state of the UserEvent Component", () => {
    const wrapper = shallow(<UserEvent match={match} />);
    wrapper.find("#address").simulate("change", {
      target: { name: "address", value: "test address" }
    });

    expect(wrapper.state("address")).toMatchSnapshot("test address");
  });
});

describe("Phone input", () => {
  it("should respond to change event and change the state of the UserEvent Component", () => {
    const wrapper = shallow(<UserEvent match={match} />);
    wrapper.find("#phone").simulate("change", {
      target: { name: "phone", value: "2345556272" }
    });

    expect(wrapper.state("phone")).toMatchSnapshot("2345556272");
  });
});

describe("Location input", () => {
  it("should respond to change event and change the state of the UserEvent Component", () => {
    const wrapper = shallow(<UserEvent match={match} />);
    wrapper.find("#location").simulate("change", {
      target: { name: "location", value: "test location" }
    });

    expect(wrapper.state("location")).toMatchSnapshot("test location");
  });
});

describe("Reason input", () => {
  it("should respond to change event and change the state of the UserEvent Component", () => {
    const wrapper = shallow(<UserEvent match={match} />);
    wrapper.find("#reason").simulate("change", {
      target: { name: "reason", value: "for test units" }
    });

    expect(wrapper.state("reason")).toMatchSnapshot("for test units");
  });
});
