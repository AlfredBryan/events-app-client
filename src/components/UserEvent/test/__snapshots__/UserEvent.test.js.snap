// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address input should respond to change event and change the state of the UserEvent Component: test address 1`] = `"test address"`;

exports[`Location input should respond to change event and change the state of the UserEvent Component: test location 1`] = `"test location"`;

exports[`Name input should respond to change event and change the state of the UserEvent Component: test name 1`] = `"test name"`;

exports[`Phone input should respond to change event and change the state of the UserEvent Component: 2345556272 1`] = `"2345556272"`;

exports[`Reason input should respond to change event and change the state of the UserEvent Component: for test units 1`] = `"for test units"`;

exports[`UserEvent Component renders Address input 1`] = `1`;

exports[`UserEvent Component renders Location input 1`] = `1`;

exports[`UserEvent Component renders Name input 1`] = `1`;

exports[`UserEvent Component renders Phone input 1`] = `1`;

exports[`UserEvent Component renders Reason input 1`] = `1`;

exports[`UserEvent Component should render without throwing an error 1`] = `false`;
